### DISCUSSION TITLE
Question around detection/event clip duration

### DISCUSSION THREAD

**Discussion by badnbusy** [blakeblackshear/frigate#5767] [General]
I have noticed that when Fridate detected an object (say a person walking by), it sometimes clips the recording to only the part where they walk by, but sometimes it records minutes before/after. I think it might be to do with other objects also occurring in the timeframe, but I don't understand why as they are also recorded as their own clip.

For example:

![image](https://user-images.githubusercontent.com/11709354/226100753-6076c5e4-b3db-4f66-8c25-16a52a61c92d.png)

The detection of the "person" happened after watching around 2 mins of the clip before they appeared. But before that, a car went by, but that event was also it's own clip. So why such a long clip length for this person?

The snapshot was:

![image](https://user-images.githubusercontent.com/11709354/226100795-e737cfae-3600-4db9-ad62-ecb0d42afe6b.png)

I can upload the full clip if it helps.

So a further question, why did it only detect the person at that point and not a few seconds earlier when they were in much clearer view right outside the driveway instead of further down the road?

Apologies for the questions, I'm still learning 😄 

Thanks.

**Comment by NickM-27** [comment]
If an event for a person is lasting much longer than the person was actually there, this is likely because frigate is seeing a person when there is none. This is considered a false positive. You'll need to view the live debug view to see where this false positive is being detected.

**Reply by badnbusy** [reply]
Thanks. So it is expected that the clip should only be for 10-20 seconds for a standard detection, rather than minutes?

**Reply by NickM-27** [reply]
It's expected that the clip for an object lasts as long as the object is visible (retention policy for record affects this somewhat).

**Reply by badnbusy** [reply]
Thanks. When recording 24/7, are these "clips" separate, or does it just use the actual 2/7/ recording file but set a start/end date when showing the clip? i.e. is it duplicating footage?

**Reply by NickM-27** [reply]
All recordings are saved as 10 second segments which are dynamically combined to show recordings and event clips

**Reply by badnbusy** [reply]
Thanks!
